### PR TITLE
[Experimental Feature] Alternate App Icons

### PR DESCRIPTION
**Pull request by LitRitt**
Mark the type contribution you are making:

- [x] Experimental feature (new functionality that can be selectively enabled/disabled)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Description

Adds assets for alternate icons, an extension to retrieve icon images as `UIImage`s, code to handle changing app icon, and feature / options.

# Testing

- iPhone 12 Pro, iOS 16.5.1
- iPad mini 6, iOS 16.5.1

# Checklist
**General (All PRs)**
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I've tested my changes with different device + OS version configurations

**Experimental Feature-specific** 
- [X] Added property to `ExperimentalFeatures` struct annotated with `@Feature`
- [X] Uses `@Option`'s to persist all feature-related data
- [X] Locked *all* behavior changes behind `ExperimentalFeatures.shared.[feature].isEnabled` runtime check
- [X] Isolates changes to separate files as much as possible (e.g. via Swift extensions)

### CODE REVIEW

File: Delta/Experimental Features/Features/AlternateAppIcons.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    
+    var id: String {
+        return self.rawValue
+    }
+    
+    var author: String {
+        switch self
+        {
+        case .normal: return "Paul Thorsen"
+        case .inverted, .gba4ios, .skin, .pixelated: return "LitRitt"
```
Comment (rileytestut) [inline Delta/Experimental Features/Features/AlternateAppIcons.swift:33]:
```suggestion
        case .normal: return "Caroline Moore"
        case .gba4ios: return "Paul Thorsen"
        case .inverted, .skin, .pixelated: return "LitRitt"
```

Caroline made the default Delta icon, whereas Paul made the GBA4iOS icon :)

**Review by rileytestut — CHANGES_REQUESTED** [review]
Awesome work, this will be a great feature to have! My only real note is a request to move the actual app-icon-changing logic out of `SettingsViewController`, but otherwise the code itself + logic all look good to me 👍

File: Delta/Settings/Settings.swift [hunk 2]
```diff
@@ -50,6 +50,7 @@ extension Settings
 struct Settings
 {
     static let features = Features.shared
+    static let experimentalFeatures = ExperimentalFeatures.shared
```
[the remark this answers is not in the document]
Comment (LitRitt) [reply Delta/Settings/Settings.swift]: Got it. Changed that.

**Comment by LitRitt** [comment]
Addressed all your comments and suggestions. All the code is now located within `AlternateAppIcons.swift`

**Pull request closed without merging**